python/taichi/core/record.py

```diff
@@ -0,0 +1,61 @@
+from .util import ti_core
+import os
+
+
+def record_action_hint(name, content=None):
+    if content is None:
+        name, content = 'hint', name
+    ti_core.record_action_hint(name, content)
+
+
+def start_recording(filename):
+    ti_core.start_recording(filename)
+
+
+def stop_recording():
+    ti_core.stop_recording()
+
+
+class RecordAction:
+    def __init__(self, filename):
+        self.filename = filename
+
+    def __enter__(self):
+        start_recording(self.filename)
+        return self
+
+    def __exit__(self, *args):
+        stop_recording()
+
+
+class RecordGroupHint:
+    def __init__(self, name):
+        if name in recorded:
```

Comment:
I am not sure if this is correct. Maybe you should use `RecordGroupHint.record` or `self.record` here.